fabrics.c

```diff
@@ -320,6 +320,27 @@ static bool ctrl_matches_connectargs(char *name, struct connect_args *args)
 	cargs.trsvcid = parse_conn_arg(addr, ' ', conarg_trsvcid);
 	cargs.host_traddr = parse_conn_arg(addr, ' ', conarg_host_traddr);
 
+	if (!strcmp(cargs.subsysnqn, NVME_DISC_SUBSYS_NAME)) {
+		char *kato_str = nvme_get_ctrl_attr(path, "kato"), *p;
+		unsigned int kato = 0;
+
+		/*
+		 * When looking up discovery controllers we have to skip
+		 * any non-persistent controllers (ie those with a zero
+		 * kato value). Otherwise the controller will vanish from
+		 * underneath us as they are owned by another program.
+		 *
+		 * The 'kato' attribute might not be present; assume a
+		 * non-persistent controller for these installations.
+		 */
```

Comment:
If the device string was set by the user on the command line and there is no `kato` attribute, we should assume that the controller is usable (i.e. will persist at least until we try to retrieve the discovery page). Otherwise you basically disable the `--device` options on kernels that don't support `kato`.

So, this logic only makes sense to me if this function is called from `find_ctrl_with_connectargs()`. But OTOH, why not generally assume that the controller is persistent if there is no `kato` attribute? The worst thing that can happen is an error attempting to connect to it, and this error will be transient by definition.